tsl/test/expected/continuous_aggs_union_view.out

```diff
@@ -0,0 +1,275 @@
+INSERT INTO metrics(time, device_id, value) SELECT '2000-01-01'::timestamptz, 1, 0.5;
+INSERT INTO metrics(time, device_id, value) SELECT '2000-01-01'::timestamptz, 2, 0.5;
+INSERT INTO metrics(time, device_id, value) SELECT '2000-01-01'::timestamptz, 3, 0.5;
+--
+-- test switching continuous agg view between different modes
+--
+-- check default view for new continuous aggregate
+CREATE VIEW metrics_summary WITH (timescaledb.continuous) AS SELECT time_bucket('1d',time), avg(value) FROM metrics GROUP BY 1;
+ALTER TABLE metrics DROP COLUMN f2;
+-- this should be union view
+SELECT pg_get_viewdef('metrics_summary',true);
+                                                                                          pg_get_viewdef                                                                                           
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+  SELECT _materialized_hypertable_2.time_bucket,                                                                                                                                                  +
+     _timescaledb_internal.finalize_agg('avg(double precision)'::text, NULL::name, NULL::name, '{{pg_catalog,float8}}'::name[], _materialized_hypertable_2.agg_2_2, NULL::double precision) AS avg+
+    FROM _timescaledb_internal._materialized_hypertable_2                                                                                                                                         +
+   WHERE _materialized_hypertable_2.time_bucket < COALESCE(_timescaledb_internal.to_timestamp(_timescaledb_internal.cagg_watermark(1)), '-infinity'::timestamp with time zone)                    +
+   GROUP BY _materialized_hypertable_2.time_bucket                                                                                                                                                +
+ UNION ALL                                                                                                                                                                                        +
+  SELECT time_bucket('@ 1 day'::interval, metrics."time") AS time_bucket,                                                                                                                         +
+     avg(metrics.value) AS avg                                                                                                                                                                    +
+    FROM metrics                                                                                                                                                                                  +
+   WHERE metrics."time" > COALESCE(_timescaledb_internal.to_timestamp(_timescaledb_internal.cagg_watermark(1)), '-infinity'::timestamp with time zone)                                            +
```

Comment:
My intuition is to expect that the time clause would be inclusive here if it was exclusive for the agg query above, or vice versa.

In other words, you'd want one query to be `<` and the other one  `>=`. Or am I missing something?

It would be good with a value in the table that exactly hits `cagg_watermark` to test this, if possible.

Reply:
your intuition was right and the ops are adjusted now